app/src/main/kotlin/com/thundermaps/apilib/android/impl/resources/MeResourceImpl.kt

```diff
@@ -169,6 +170,17 @@ class MeResourceImpl @Inject constructor(
         return resultHandler.processResult(call, gson)
     }
 
+    override suspend fun updateGdpr(
+        parameters: RequestParameters,
+        updateGdprBody: UpdateGdprBody
+    ): Result<Unit> {
+        if (!parameters.host.isInternetAvailable()) {
+            return resultHandler.handleException(UnknownHostException())
```

Comment:
Should this be handled with another exception ?

Reply:
This returns a Result<Error> type, it's a indication of a error in API, but not a java exception.